### ISSUE TITLE
SlimHUD sometimes fails to suppress the macOS HUDs

### ISSUE THREAD

**Issue by Ein-Tim** [AlexPerathoner/SlimHUD#134]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Updated SlimHUD to the latest version (if applicable)

### Describe the bug

I'm sometimes seeing two HUDs with SlimHUD installed, on one side the original macOS HUD and then the one from SlimHUD.

### Steps to reproduce

Not entirely clear as this doesn't happen always.

### Expected behavior

While SlimHUD is running, the macOS HUDs should be suppressed.

### Environment Information (please complete the following information)

```markdown
- macOS version: 13.3
- Mac model: MacBook Air
- Monitor(s): internal
- Apple Silicon/M1 (yes or no): yes
```

**Comment by Ein-Tim** [comment]
Here's a screenshot: <img width="1440" alt="Bildschirmfoto 2023-03-30 um 22 23 33" src="https://user-images.githubusercontent.com/67682506/228955614-ed767215-4461-4942-80ca-4be2ad0e9745.png">

**Comment by AlexPerathoner** [comment]
Does this happen immediately after startup? To solve it should be enough to restart the app, I'll investigate in the meantime

**Comment by Ein-Tim** [comment]
@AlexPerathoner Yes, restarting is a temporary workaround, but I noticed that the time frame between App restart and both HUDs being shown decreased with every app restart. So a real fix for this would be great, thanks!

**Comment by AlexPerathoner** [comment]
Oh it happens after the app has started and the huds had been hidden already?

**Comment by nkean** [comment]
Are they reappearing randomly?  I've noticed it only happens after I wake my laptop from sleep.

**Comment by AlexPerathoner** [comment]
@nkean that would at least make sense. macOS's launchd could be noticing that OSDUIHelper has stopped responding, and restarted it.
However, with newer releases of macOS, notifications warning about launch-at-login processes appear at random time too...
If either of those is the case, it could be necessary to constantly check if OSDUI is running...

**Comment by Ein-Tim** [comment]
@nkean I tried to reproduce the bug by closing my MacBook and awaking it from sleep ca. 8 hours later, but at the moment SlimHUD still works correctly and the macOS HUDs are correctly suppressed.

I will continue trying to find a pattern here.

**Comment by carsuki** [comment]
Also seeing this issue but at least in my case it seems like sleep isn't even a factor (at least not the main one). After a while of prolonged laptop use I'll just sometimes notice it starting out of nowhere next time I adjust the volume.

**Comment by chrisfrantz** [comment]
I'm having the same issue, M1 MacBook Pro on Ventura. It only ever works on restart for a few uses before falling back to showing slimhud and the default hud simultaneously.

**Comment by dhru7en** [comment]
Same issue here on M1 Air Monterey. Even restarting the app or restarting my Mac doesn't resolve it.

**Comment by RubikOwl** [comment]
I am having this issue too, and while troubleshooting I quit the app from the dock. But then when I went into Activity monitor, it still showed up as running. Quitting it there and restarting the app seemed to fix it and hide the default mac HUD, but it does come back sometimes, so I just force quit SlimHUD again. Not sure if that helps anyone, but thought I'd throw in my 2 cents while I was here.

**Comment by loyalOrange503** [comment]
having the same issue here, on the latest version of big sur

**Comment by RubikOwl** [comment]
As a quick tip, turning on "Continuous check" does seem to make it take longer to break. Hope that helps, at least in making it less annoying till a fix is found.

**Comment by FormalSnake** [comment]
Continuous check breaks it completely for me.

**Comment by loyalOrange503** [comment]
> As a quick tip, turning on "Continuous check" does seem to make it take longer to break. Hope that helps, at least in making it less annoying till a fix is found.

This has helped to an extend, thanks!

I would like to add, I was just watching a video in quicktime player, once the video finished the MacOS volume indicator showed up without me touching any of the volume buttons. it disappeared less then a second later, however, after this the MacOS HUD was no longer suppressed, so it might have to do something with switching applications, or starting or finishing playing audio.

**Comment by alucardontop** [comment]
same issue, ventura 13.4.1 and m1 max

**Comment by doxadeo** [comment]
I also have the same issue with MacOS HUD appearing again after some time. It does seem a bit random how long it takes to start showing the MacOS HUD again.

I created a workaround using Shortcuts to quit and reopen SlimHUD, and assigned a keyboard shortcut, so I can quickly restart SlimHUD. Sometimes SlimHUD doesn't reopen and I need to press the shortcut again. Not ideal, but it works.

**Comment by FormalSnake** [comment]
Sometimes it suppresses but it doesn't show anything

**Comment by omarsonmarz** [comment]
Same issue, M1 MBP w/Touch Bar and on sonoma public beta

**Comment by FormalSnake** [comment]
still a problem indeed

**Comment by RubikOwl** [comment]
TIP: So earlier in this tread I said that turning on "Continuous Check" seems to help the issue. In my usage since then, I've found that it actually works in EITHER mode, but only for a while. Eventually you need to toggle it on or off and that seems to kick it enough to fix it for another while. Restarting SlimHUD by force quitting it does still seem to also temporarily fix it, toggling continuous check should be done if restarting the app doesn't seem to be working.

**Comment by tommy-mitchell** [comment]
I was running into this issue every time my Mac woke from sleep. I installed [sleepwatcher](https://formulae.brew.sh/formula/sleepwatcher) and added this to `~/.wakeup`:

```sh
#!/usr/bin/env bash

killall SlimHUD;
open -a SlimHUD;
```

Works like a charm now. Follow the homebrew installation instructions and make the [sleepwatcher scripts executable](https://apple.stackexchange.com/questions/457292/sleepwatcher-on-ventura):

```sh
chmod a+x ~/.sleep
chmod a+x ~/.wakeup
```

**Comment by Stargorithm** [comment]
This is happening to me as well on m1 air running macos sonoma

**Comment by benjamintxn** [comment]
Same here, on Sonoma

**Comment by Bellavene** [comment]
`launchctl unload -F /System/Library/LaunchAgents/com.apple.OSDUIHelper.plist > /dev/null 2>&1 &`

**Comment by AlexPerathoner** [comment]
I've released a beta release that might fix this. In the Info tab in the bottom right corner there's a "Join the Beta" box. Please try that out and let me know if it's fixing the issue.

**Comment by Bellavene** [comment]
Thanks for the downvote. I am glad that you enjoyed at least the hook description without votes :D

**Comment by AlexPerathoner** [comment]
mb lol, downnote was a bit aggressive without any exaplation. `launchctl unload` would do anything on my machine, therefore the 👎 

Hooks however seem to be working well, thanks for the tip!

**Comment by Bellavene** [comment]
This command speaks for itself, it will turn off built in OSD, right what OP was asking for :D
